openssl/src/ssl/mod.rs

```diff
@@ -1437,8 +1437,9 @@ impl SslContextBuilder {
     /// The callback will be called with the SSL context and a slice into which the cookie
     /// should be written. The callback should return the number of bytes written.
     ///
-    /// This corresponds to `SSL_CTX_set_cookie_generate_cb`.
-    pub fn set_cookie_generate_cb<F>(&mut self, callback: F)
+    /// This corresponds to `SSL_CTX_set_stateless_cookie_generate_cb`.
+    #[cfg(ossl111)]
```

Comment:
Why is this no longer supported on older versions?

Reply:
`SSL_CTX_set_stateless_cookie_generate_cb` is new in OpenSSL 1.1.1-pre3. In -pre1, a DTLSv1-related interface was overloaded, but this was deemed suboptimal.

Reply:
So these methods are already part of the public API, so I don't think we can get rid of them. Seems fine to have both the old DTLSv1 and new TLSv1.3 stuff at the same time though? https://docs.rs/openssl/0.10.5/openssl/ssl/struct.SslContextBuilder.html#method.set_cookie_verify_cb

Reply:
Easy enough to support both here, but in general breaking changes between pre-release versions are expected.

Reply:
Someone using `set_cookie_verify_cb` and linking against OpenSSL 1.0.2 or 1.1.0 isn't using a prerelease version of anything AFAIKT.

Reply:
Fair enough. Anyway, it's all there now.